cltk/corpus/latin/wordnet.py

```diff
@@ -0,0 +1,2291 @@
+# -*- coding: utf-8 -*-
+
+
+"""
+A CLTK interface for Latin WordNet, built on the NLTK WordNet API
+
+Latin WordNet is a lexico-semantic database of Latin.
+Using synsets, helps find conceptual relationships between words
+such as hypernyms, hyponyms, synonyms, antonyms etc.
+
+"""
+
+from __future__ import print_function, unicode_literals
+
+import codecs
+import math
+import os
+import re
+import string
+from collections import defaultdict, deque
+from functools import total_ordering
+from itertools import chain
+from operator import itemgetter
+
+import requests
+from nltk.compat import python_2_unicode_compatible
+from nltk.corpus.reader import CorpusReader
+from nltk.probability import FreqDist
+from six import iteritems
+
+from cltk import get_cltk_data_dir
+
+
+nesteddict = lambda: defaultdict(nesteddict)
+punctuation = str.maketrans("", "", string.punctuation)
+
+######################################################################
+# Table of Contents
+######################################################################
+# - Constants
+# - Data Classes
+#   - WordNetError
+#   - Lemma
+#   - Synset
+# - WordNet Corpus Reader
+# - WordNet Information Content Corpus Reader
+# - Similarity Metrics
+# - Demo
+
+######################################################################
+# Constants
+######################################################################
+
+#: Positive infinity (for similarity functions)
+_INF = 1e300
+
+# { Part-of-speech constants
+ADJ, ADV, NOUN, VERB, PREP = "a", "r", "n", "v", "p"
+# }
+
+POS_LIST = [NOUN, VERB, ADJ, ADV, PREP]
+
+SENSENUM_RE = re.compile(r"^([nvarp])#(\w+)$")
+
+
+######################################################################
+# Data Classes
+######################################################################
+
+
+class WordNetError(Exception):
+    """An exception class for wordnet-related errors."""
+
+
+class _WordNetObject(object):
+    """A common base class for lemmas and synsets."""
+
+    def antonyms(self):
+        """
+        >>> LWN = WordNetCorpusReader()
```

Comment:
General comment: I see you've added A TON of tests. THANK YOU, it makes all the difference for the long term support of this code.